sat/cli/bootprep/input/configuration.py

```diff
@@ -303,41 +309,44 @@ def commit(self):
         return self.matching_product.commit
 
 
-class InputConfiguration:
+class InputConfiguration(BaseInputItem):
     """A CFS Configuration from a bootprep input file."""
+    description = 'CFS configuration'
 
-    create_error_cls = ConfigurationCreateError
+    # Use InputItemValidateError since fields are rendered in validation methods.
+    # TODO: Consider renaming to something like `template_render_err` because this is
```

Comment:
> Kind of related to this comment - we're currently setting BaseInputImage.create_error_cls to ImageCreateError. Is there any point to that?

Yes, it is quite confusing though. There are a few properties in `BaseInputImage` that use the `jinja_rendered` decorator. That decorator pulls the value of the class attribute `create_error_cls` and raises that if the Jinja template render fails. Those properties get rendered in the `sat.cli.bootprep.image.create_images` function, and that function is enclosed in a `try` `except ImageCreateError`. I could change this, but I'm not planning to right now.

> If we eventually refactor InputImage to be a subclass of BaseInputItem would it need to have a different create_error_cls?

Yes, most likely we would want it to be the `InputItemValidationError`. and then at that point, it probably makes sense to just use the default or perhaps just get rid of the class attribute altogether since it's a little too subtle and confusing.

> Also, InputItemValidateError is the default value for BaseInputItem so not strictly needed here.
>
> Not sure it's a big deal but I think it'd be marginally clearer to remove it from all the subclasses and maybe put this TODO comment into BaseInputItem. Maybe that would make sense to do when the TODO is addressed, or when InputImage gets refactored.

Yeah, I think I'll remove the `create_error_cls` from both the `InputSessionTemplate` and `InputConfiguration` classes at least since they can both use the default. As described above though, I can't do that for the `BaseInputImage` yet.

Reply:
Oh, one minor additional caveat: the `InputConfigurationLayer` class also sets the `create_error_cls` although it doesn't use the `BaseInputItem` as a base class. It sets it to `ConfigurationCreateError` because the rendering of the fields in the CFS layers only occurs once we're creating the CFS configurations. It doesn't really have to be that way, as far as I can tell, but that's the way it is for now.